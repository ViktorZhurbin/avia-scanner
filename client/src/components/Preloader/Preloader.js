import React from 'react';
import cl from 'classnames/bind';
import PropTypes from 'prop-types';

import styles from './Preloader.css';

const cx = cl.bind(styles);

const Preloader = ({ className, size, color }) => (
    <span className={cx('preloader', className)}>
        <i className={cx('wheel', `color-${color}`, {
            s: size === 's',
            m: size === 'm'
        })}
        />
    </span>
);

Preloader.propTypes = {
    size: PropTypes.oneOf(['s', 'm']),
    color: PropTypes.oneOf([
        'metal',
        'white',
        'black',
        'twilight'
    ]),
    className: PropTypes.string
};

Preloader.defaultProps = {
    size: 's',
    color: 'white',
    className: ''
};

export default React.memo(Preloader);
