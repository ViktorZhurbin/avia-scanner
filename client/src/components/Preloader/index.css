@keyframes spin {
    from {
        transform: rotate(0deg);
    }

    to {
        transform: rotate(360deg);
    }
}

.preloader {
    display: inline-block;
    line-height: 32px;
    padding: 9px 0;
}

.wheel {
    display: inline-block;
    box-sizing: border-box;
    margin: 0;
    animation: spin 1s infinite linear;
    border-radius: 50%;
    font: inherit;

    &.color-white {
        border: 2px solid var(--color-white);
        border-top-color: transparent;
        border-left-color: transparent;
    }
}

.wheel::before {
    visibility: hidden;
    content: '\A0';
}

.s {
    height: 30px;
    width: 30px;
}

.m {
    height: 48px;
    width: 48px;
}
