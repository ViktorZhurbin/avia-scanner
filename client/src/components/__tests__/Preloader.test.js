import React from 'react';
import renderer from 'react-test-renderer';

import Preloader from '../Preloader';

describe('Render Preloader', () => {
    it('render correctly', () => {
        const component = renderer.create(
            <Preloader />,
        );
        expect(component.toJSON()).toMatchSnapshot();
    });
});
