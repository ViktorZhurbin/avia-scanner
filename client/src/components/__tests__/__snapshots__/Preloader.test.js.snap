// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Preloader render correctly 1`] = `
<span
  className="preloader"
>
  <i
    className="wheel color-white s"
  />
</span>
`;
